src/Microsoft.Azure.EventHubs/Amqp/AmqpMessageConverter.cs

```diff
@@ -47,7 +47,7 @@ public static AmqpMessage EventDatasToAmqpMessage(IEnumerable<EventData> eventDa
         {
             AmqpMessage returnMessage = null;
             int dataCount = eventDatas.Count();
```

Comment:
>dataCount [](start = 16, length = 9)

Good catch. We should just check the dataCount here. eventDatas should never be null. Please throw if it is.